### PR TITLE
1559133: Make landing page less confusing

### PR DESCRIPTION
**Pull request by mdboom**
This should make the docs make sense whether the reader is coming for glean-ac or glean-core.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

**Comment by badboy** [comment]
Network issues on CircleCI caused failures.

File: README.md [hunk 1]
```diff
@@ -20,7 +12,12 @@ All documentation is available online:
 
 ## Overview
 
-This repository is used to build the client-side cross-platform Telemetry library part of [Glean](https://docs.telemetry.mozilla.org/concepts/glean/glean.html), called the `Glean SDK`.
+There currently two independent implementations of the Glean SDK:
```
Comment (Dexterp37) [inline README.md]: I would put this at the end of this section, rather than mentioning it upfront

File: README.md [hunk 2]
```diff
@@ -35,14 +32,12 @@ This repository also hosts the documentation for the `Glean SDK`:
...
 
-For an overview of Glean, see the [section in the Firefox data docs](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).
-
 ## Contact
 
 To contact us you can:
 
 * Find us on the Mozilla Slack in *#glean*, on [Mozilla IRC][mozirc] in *#telemetry*.
-* To report issues or request changes, file a bug in [Bugzilla in Data Platform & Tools :: Glean: SDK][newbugzilla].
+* To report issues or request changes, file a bug in [Bugzilla in Data Platform & Tools :: Glean: SDK](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data%20Platform%20and%20Tools&component=Glean%3A%20SDK).
```
Comment (Dexterp37) [inline README.md]: Mh, let's leave the old link there (it's referencing a link at the end of this file) for now: we need to decide how to handle the two triages and how to file bugs for the two different products.

File: README.md (hunk 1 above)
Comment (mdboom) [reply README.md]: I had considered that, but then the subsequent paragraphs don't have any context.  I've reorganized this more substantially.  Let me know what you think.

Comment (Dexterp37) [reply README.md]: Thank you, looks good to me now!

**Pull request merged**